src/gui/settings/qgssettingseditorwidgetwrapperimpl.cpp

```diff
@@ -81,6 +81,75 @@ QString QgsSettingsStringLineEditWrapper::valueFromWidget() const
   return QString();
 }
 
+// *******
+// String with combo box
+// *******
+
+QString QgsSettingsStringComboBoxWrapper::id() const
+{
+  return QString::fromUtf8( sSettingsTypeMetaEnum.valueToKey( static_cast<int>( Qgis::SettingsType::String ) ) );
+}
+
+bool QgsSettingsStringComboBoxWrapper::setWidgetValue( const QString &value ) const
+{
+  if ( mEditor )
+  {
+    int idx = mMode == Mode::Data ? mEditor->findData( value ) : mEditor->findText( value );
+    if ( idx >= 0 )
+    {
+      mEditor->setCurrentIndex( idx );
+      return true;
+    }
+    else
+    {
+      return false;
+    }
+  }
+  else
+  {
+    QgsDebugError( QStringLiteral( "Settings editor not set for %1" ).arg( mSetting->definitionKey() ) );
```

Comment:
How bad is this? Does it deserve a `Q_ASSERT` (i.e. crash on debug builds)?

Reply:
I don't think. If this is badly used in plugins (using the wrapper with an invalid editor), we don't want to crash.